beetsplug/replaygain.py

```diff
@@ -312,10 +327,25 @@ def end_element_handler(name):
                                           'the output of bs1770gain')
                 album_gain["album"] = Gain(state['gain'], state['peak'])
                 state['gain'] = state['peak'] = None
+            elif len(per_file_gain) == len(path_list):
+                if state['gain'] is not None:
+                    album_state['gain'] = state['gain']
+                if state['peak'] is not None:
+                    album_state['peak'] = state['peak']
+                if album_state['gain'] is not None \
+                        and album_state['peak'] is not None:
+                    album_gain["album"] = Gain(
+                        album_state['gain'], album_state['peak'])
+                state['gain'] = state['peak'] = None
+
         parser.StartElementHandler = start_element_handler
         parser.EndElementHandler = end_element_handler
 
         try:
+            if type(text) == bytes:
+                text = text.decode('utf-8')
+            while '\x08' in text:
+                text = re.sub('[^\x08]\x08', '', text)
```

Comment:
This replacement is somewhat surprising! Maybe a comment would be useful here to describe what’s going wrong?